daemon/algod/api/server/v1/routes/routes.go

```diff
@@ -38,7 +36,7 @@ var Routes = lib.Routes{
 	lib.Route{
 		Name:        "wait-for-block",
 		Method:      "GET",
-		Path:        "/status/wait-for-block-after/{round:[0-9]+}",
+		Path:        "/status/wait-for-block-after/:round",
```

Comment:
It looks as if we have lost the ability to do a regex sanitation over the input arguments, which could lead to undesired consequences.
I think that you'd like to retain that. Suggestion : When registering each of these entry points, direct that to a stub function which would perform the sanitation. if all good, it could be forwarded to the underlying implementation.

Reply:
We were already validating these in the code, so removing the regex didn't really effect anything.

Reply:
We are validating strings which were previously pruned. Now we're not.
I don't know if the existing string validation is sufficiently "smart" to handle these use cases in the same way. ( do I want it to be ? yes. is it ? I'm not really sure )
for instance, the parser might be now capable of parsing round value of 0x60, whereas it didn't before.

Reply:
I see, I think that the code we already had to verify these thins is stricter than the regex but maybe I'm incorrect.

My code is now depending on this for round:
```
	queryRound, err := strconv.ParseUint(context.Param("round"), 10, 64)
	if err != nil {
		lib.ErrorResponse(w, http.StatusBadRequest, err, errFailedParsingRoundNumber, ctx.Log)
		return
	}
```

this for address:
```
	queryAddr := context.Param("addr")
	addr, err := basics.UnmarshalChecksumAddress(queryAddr)
	if err != nil {
		lib.ErrorResponse(w, http.StatusBadRequest, err, errFailedToParseAddress, ctx.Log)
		return
	}
```

and this for txid:
```
	queryTxID := context.Param("txid")
	if queryTxID == "" {
		lib.ErrorResponse(w, http.StatusBadRequest, fmt.Errorf(errNoTxnSpecified), errNoTxnSpecified, ctx.Log)
		return
	}

	txID := transactions.Txid{}
	if txID.UnmarshalText([]byte(queryTxID)) != nil {
		lib.ErrorResponse(w, http.StatusBadRequest, fmt.Errorf(errNoTxnSpecified), errNoTxnSpecified, ctx.Log)
		return
	}
```

Reply:
It is actually better to not have the regex and it something I wanted to fix in this version. The issue is from the user experience. Since the rejection happens in the routing layer, the user gets the wrong error. For example, if  the user hit `/v2/blocks/345khj`, the user would a get a `404` (because the regex prevented it from matching any address) rather than 400 with sensible "please correct your block number" error message.

Reply:
Yes, that's a good point. My concern is that we need now to revalidate all the entry point parsers.

Reply:
@tsachiherman Great, sounds like we are all on the same page here. When making this change I went through the exercise of verifying that each spot where I replaced the regex validation was properly validated in the code.